packages/lib/reducer.test.js

```diff
@@ -590,6 +590,45 @@ describe('reducer', function() {
 		expect(state.selectedNoteIds[0]).toEqual(notes[1].id);
 	});
 
+	it('should record last selected notes when all notes filter is on', async () => {
+		const folders = await createNTestFolders(2);
+		const notes = [];
+		for (let i = 0; i < folders.length; i++) {
+			notes.push(...await createNTestNotes(2, folders[i]));
+		}
+
+		// Initialize state with selected note at 0 index
+		let state = initTestState(folders, [0], notes.slice(), [0]);
+
+		// Turn on 'All Notes' filter
+		state = reducer(state, { type: 'SMART_FILTER_SELECT', id: ALL_NOTES_FILTER_ID });
+
+		// Building the history to test HISTORY_BACKWARD and HISTORY_FORWARD when All Notes filter is on.
+		state = goToNote(notes, [3], state);
+		state = goToNote(notes, [1], state);
+		state = goToNote(notes, [2], state);
+
+		expect(state.selectedNoteIds[0]).toEqual(notes[2].id);
+
+		state = goBackWard(state);
+		expect(state.selectedNoteIds[0]).toEqual(notes[1].id);
+
+		state = goBackWard(state);
+		expect(state.selectedNoteIds[0]).toEqual(notes[3].id);
+
+		state = goBackWard(state);
+		expect(state.selectedNoteIds[0]).toEqual(notes[0].id);
+
+		state = goForward(state);
+		expect(state.selectedNoteIds[0]).toEqual(notes[3].id);
+
+		state = goForward(state);
+		expect(state.selectedNoteIds[0]).toEqual(notes[1].id);
+
+		state = goForward(state);
+		expect(state.selectedNoteIds[0]).toEqual(notes[2].id);
+	});
+
```

Comment:
That feels like a very long test with lots of repetitions. What would be the minimum test to verify that your change is correct?

Reply:
This test was in the PR https://github.com/laurent22/joplin/pull/4990. The simple test which can verify my change is correct is 1) Select a Note 2) Select All Notes smart filter 3) Add tag to current note 4) The selected note should not default to first on the list as it was before the bug fix

Reply:
How do i refresh notes in test file?